### PR TITLE
HIVE-25659: Metastore direct sql queries with IN/(NOT IN) should be split based on max parameters allowed by SQL DB.

### PR DESCRIPTION
**Pull request by guptanikhil007**
### What changes were proposed in this pull request?

This PR adds a new config to help in division of queries. It will help in creating queries that are compatible with sql max parameters limit in MS SQL.

### Why are the changes needed?

The default configurations can create queries with very huge number of parameters in the query.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Unit Tests

### CODE REVIEW

**Review by adesh-rao — APPROVED** [review]

File: standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/txn/TestTxnUtils.java [hunk 1]
```diff
@@ -150,12 +151,34 @@ public void testBuildQueryWithINClause() throws Exception {
...
-    Assert.assertEquals(2033L, ret.get(1).longValue());
-    Assert.assertEquals(33L, ret.get(2).longValue());
+    Assert.assertEquals(2000L, ret.get(0).longValue());
+    Assert.assertEquals(2000L, ret.get(1).longValue());
+    Assert.assertEquals(321L, ret.get(2).longValue());
     runAgainstDerby(queries);
   }
 
+  @Test(expected = IllegalArgumentException.class)
+  public void testBuildQueryWithNOTINClauseFailure() throws Exception {
```
Comment (sankarh) [inline standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/txn/TestTxnUtils.java:161]: Why would this test fail?

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnUtils.java [hunk 2]
```diff
@@ -316,7 +317,7 @@ public static String getFullTableName(String dbName, String tableName) {
       // Compute the size of a query when the 'nextValue' is added to the current query.
       int querySize = querySizeExpected(buf.length(), nextValue.length(), suffix.length(), addParens);
 
-      if (querySize > maxQueryLength * 1024) {
+      if (querySize > maxQueryLength * 1024 || currentCount >= maxParameters) {
```
Comment (sankarh) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnUtils.java:320]: nit: Keep each condition within ()

File: standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java [hunk 3]
```diff
@@ -680,6 +680,9 @@ public static ConfVars getMetaConf(String name) {
     DIRECT_SQL_MAX_ELEMENTS_VALUES_CLAUSE("metastore.direct.sql.max.elements.values.clause",
         "hive.direct.sql.max.elements.values.clause",
         1000, "The maximum number of values in a VALUES clause for INSERT statement."),
+    DIRECT_SQL_MAX_PARAMETERS("metastore.direct.sql.max.parameters",
+        "hive.direct.sql.max.parameters", 1000, "The maximum parameters the\n" +
```
[the remark this answers is not in the document]
Comment (sankarh) [reply standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java:684]: Rephrase: "... maximum query parameters..."

File: standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/txn/TestTxnUtils.java (hunk 1 above)
Comment (guptanikhil007) [reply standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/txn/TestTxnUtils.java:161]: If the not in query cannot be fit in one query, it should fail according to function definition

**Pull request merged**